src/main/scala/nl.knaw.dans.easy.properties/app/repository/sql/QueryGenerator.scala

```diff
@@ -103,30 +108,54 @@ object QueryGenerator {
       filters.isNewVersionFilter.map(createSearchSimplePropertiesSubQuery(_)("is-new-version", filter => BooleanUtils.toStringTrueFalse(filter.isNewVersion))),
       filters.curationRequiredFilter.map(createSearchSimplePropertiesSubQuery(_)("is-curation-required", filter => BooleanUtils.toStringTrueFalse(filter.curationRequired))),
       filters.curationPerformedFilter.map(createSearchSimplePropertiesSubQuery(_)("is-curation-performed", filter => BooleanUtils.toStringTrueFalse(filter.curationPerformed))),
-      filters.contentTypeFilter.map(createSearchSimplePropertiesSubQuery(_)("content-type", _.value.toString)),
+      filters.contentTypeFilter.map(createSearchSimplePropertiesSubQuery(_)("content-type", _.value)),
     )
       .collect {
-        case Some((tableName, q, values)) if queryWherePart.isEmpty => s"INNER JOIN ($q) AS ${ tableName }SearchResult ON Deposit.depositId = ${ tableName }SearchResult.depositId" -> values
-        case Some((tableName, q, values)) => s"INNER JOIN ($q) AS ${ tableName }SearchResult ON SelectedDeposits.depositId = ${ tableName }SearchResult.depositId" -> values
-      }
-      .foldLeft(("", List.empty[PrepStatementResolver])) {
-        case (("", vs), (subQuery, values)) => subQuery -> (values ::: vs)
-        case ((q, vs), (subQuery, values)) => s"$q $subQuery" -> (values ::: vs)
+        case Some((tableName, q, resolvers)) =>
+          (depositTableName: String) => s"INNER JOIN ($q) AS ${ tableName }SearchResult ON $depositTableName.depositId = ${ tableName }SearchResult.depositId" -> resolvers
       }
-
-    val (query, resolvers) = (queryJoinPart, queryWherePart) match {
-      case ("", "") =>
-        val query = s"SELECT * FROM Deposit"
-        query -> Nil
-      case ("", _) =>
-        val query = s"SELECT * FROM Deposit WHERE $queryWherePart"
-        query -> whereValues.reverse
-      case (_, "") =>
-        val query = s"SELECT * FROM Deposit $queryJoinPart"
-        query -> joinValues.reverse
-      case (_, _) =>
-        val query = s"SELECT * FROM (SELECT * FROM Deposit WHERE $queryWherePart) AS SelectedDeposits $queryJoinPart"
-        query -> (whereValues.reverse ::: joinValues.reverse)
+      .reduceOption((f, g) => (depositTableName: String) => {
+        val (queryWherePartF, valuesF) = f(depositTableName)
+        val (queryWherePartG, valuesG) = g(depositTableName)
+
+        s"$queryWherePartF $queryWherePartG" -> (valuesG ::: valuesF)
+      })
+
+    val baseQuery = "SELECT * FROM Deposit"
+    val (query, resolvers) = (maybeJoinPart, maybeLastModifiedPart, maybeWherePart) match {
```

Comment:
yep, only 'last modified' here. Since `creationTimestamp` is part of the `Deposit` table, this is part of the `maybeWherePart`. 'last modified' was quite a complex query and is now a `VIEW` in SQL and hence required its own particular place within the potential SQL query.